Add Dockerfile

feuGeneA opened the pull request:
I didn't like the idea of blindly running on my personal machine the bash install scripts for nix, dapptools and foundry, so I created this for my own purposes, and thought I would offer it as a contribution.

feuGeneA committed:
Add Dockerfile

mds1 commented on Dockerfile:
```diff
@@ -0,0 +1,61 @@
+# that running the image will execute benchmarks with a "warm cache". For
+# example:
+#
+#   docker --build-arg ALCHEMY_API_KEY=... -t sim .
```
I'm a docker amateur so may be doing something wrong, but running this command gives me the error below.

```
$ docker --build-arg ALCHEMY_API_KEY=<myKey> -t sim .
unknown flag: --build-arg
See 'docker --help'.
```

Using macOS 11.6.2 and `docker version` gives this output: https://app.warp.dev/block/4gBXpbdr0X86tjbkRElxqL

feuGeneA replied:
Oops! My bad! I simply made a manual error in that command. Addressed in https://github.com/mds1/convex-shutdown-simulation/pull/5/commits/f572e96fcf22aea65e41fdc111df78666d5dbec9

Sure would be nice if there were a GitHub Action or other CI thing that would exercise this thing and provide an authoritative command source :)

mds1 replied:
Awesome, this is great + working as expected now.

Good call on CI, just created https://github.com/mds1/convex-shutdown-simulation/issues/6 to track

mds1 commented on Dockerfile:
```diff
@@ -0,0 +1,61 @@
+
+FROM debian
+
+ARG ALCHEMY_API_KEY
```
Can we update this to the full `ETH_RPC_URL` instead of just overwriting the API key portion? In the `.env.template` I use alchemy as an example, but in general I prefer specifying the full RPC URL as the env var so it's easier to use any node provider / a local node

feuGeneA replied:
Addressed in https://github.com/mds1/convex-shutdown-simulation/pull/5/commits/e3d61d6691bfcd3d72907714ff00a226e648dd75

mds1 commented on Dockerfile:
```diff
@@ -0,0 +1,61 @@
+# Building this image will include the running of all benchmarks in a "cold
+# cache" state, and thus will take quite a bit of time (roughly 30-60 minutes,
+# with Ganache excluded).  The cache will be retained in the built image, such
+# that running the image will execute benchmarks with a "warm cache". For
+# example:
```
How would you clear the cache if you wanted to? Would be good to add that in the comments here

feuGeneA replied:
I've got an example command drafted for this, but I'm still verifying it. Will report back in a few hours.

mds1 replied:
Perfect, thanks! Everything else here looks good so will merge once that's added

feuGeneA replied:
Addressed in https://github.com/mds1/convex-shutdown-simulation/pull/5/commits/565d6f9b049a8128dbfbee86516c2e37b4fcf12b

mds1 reviewed (commented):
Nice, thanks for this! A few comments before merging

feuGeneA committed:
Fix example docker command

Addresses review comment https://github.com/mds1/convex-shutdown-simulation/pull/5#discussion_r788300248

feuGeneA committed:
Parameterize ETH_RPC_URL, not ALCHEMY_API_URL

Addresses review comment https://github.com/mds1/convex-shutdown-simulation/pull/5#discussion_r788300730

feuGeneA committed:
Example command to clear cache and run

Addresses review comment https://github.com/mds1/convex-shutdown-simulation/pull/5#discussion_r788304331

feuGeneA committed:
Fixup comments: RPC, not Alchemy
